miro_preprocessor/miro_inventory/src/miro_inventory.py

```diff
@@ -0,0 +1,79 @@
+# -*- encoding: utf-8 -*-
+"""
+Description
+"""
+
+import base64
+import json
+import os
+
+from botocore.vendored import requests
+
+
+# Super naive jsonpath!
+def _extract_id_field(o, id_field):
+    path = id_field.split(".")
+
+    for node in path:
+        o = o[node]
+
+    return o
+
+
+def _generate_indexable_json(event, id_field):
+    subject = event['Records'][0]['Sns']['Subject']
+    message = json.loads(event['Records'][0]['Sns']['Message'])
+    id = _extract_id_field(message, id_field)
+
+    return json.dumps({
+        'id': id,
+        'subject': subject,
+        'message': message
+    })
+
+
+def _generate_auth_token(username, password):
+    plaintext = f'{username}:{password}'
+    return base64.b64encode(plaintext.encode())
+
+
+def _post_to_es(es_cluster_url, es_index, es_type, es_username, es_password, payload):
+    token = _generate_auth_token(es_username, es_password)
+    post_url = f'{es_cluster_url}/{es_index}/{es_type}'
+
+    print(f'POSTing to {post_url}')
+
+    ascii_token = token.decode('ascii')
+
+    return requests.post(
+        post_url,
+        data=payload,
+        headers={'Authorization': f'Basic {ascii_token}'}
```

Comment:
Try:

```python
return requests.post(
   post_url,
   data=payload,
   auth=(es_username, es_password)
)
```

Because requests has builtin support for HTTP Basic Auth – is that what you’re trying to do here?